platooning_strategic/src/platoon_strategic.cpp

```diff
@@ -974,6 +1028,27 @@ namespace platoon_strategic
             pm_.memberUpdates(vehicleID, platoonId, msg.header.sender_bsm_id, statusParams, dtd);
             ROS_DEBUG_STREAM("Received platoon status message from " << msg.header.sender_id);
         }
+        else if(isJoinRequirementsMsg && msg.header.recipient_id == config_.vehicleID) {
+            has_received_join_requirements_ = true;
+
+            // JOIN_REQUIREMENTS message uses params string format "JOIN_REQUIREMENTS|LANE_INDEX:xx"
+            std::vector<std::string> inputsParams;
+            boost::algorithm::split(inputsParams, strategyParams, boost::is_any_of(","));
+
+            std::vector<std::string> target_lane_index_parsed;
+            boost::algorithm::split(target_lane_index_parsed, inputsParams[0], boost::is_any_of(":"));
+            int target_lane_index = std::stoi(target_lane_index_parsed[1]);
+
+            ROS_DEBUG_STREAM("Received JOIN_REQUIREMENTS MobilityOperation with target lane index: " << target_lane_index);
+
+            // Store the target CandidateFollower platoon lane index provided by the target Leader
+            cf_target_lane_index_ = target_lane_index;
```

Comment:
We're modifying state here, and then referencing it (potentially at a different timestep) above. Is there a possibility for issues with multiple negotiations going on simultaneously? Either multiple leaders negotiating with a single follower or multiple followers negotiating with the same leader? If so, it'd be good to add some kind of lock - almost like a mutex - where we record the ID of the vehicle we're negotiating with (if any) and don't process incoming messages from any other ID until that negotiation is completed (or maybe a timeout).

Does that sound like a realistic concern as we approach supporting 3+ vehicle platoons?

Reply:
I did find that multiple negotiations could occur simultaneously for both scenarios: (1) a leader negotiating with multiple followers, and (2) a follower receiving JOIN_REQUIREMENTS from multiple leaders. As you've suggested, I've provided checks to ensure that the full negotiation only occurs is one other vehicle (based on its ID).